src/librustdoc/clean/inline.rs

```diff
@@ -420,6 +423,20 @@ crate fn build_impl(
     if trait_.def_id() == tcx.lang_items().deref_trait() {
         super::build_deref_target_impls(cx, &trait_items, ret);
     }
+
+    // Return if the trait itself or any types of the generic parameters are doc(hidden).
+    let mut deque: VecDeque<&Type> = trait_.iter().collect();
+    while let Some(ty) = deque.pop_back() {
+        if let Some(did) = ty.def_id() {
+            if cx.tcx.get_attrs(did).lists(sym::doc).has_word(sym::hidden) {
+                return;
+            }
+        }
+        if let Some(generics) = ty.generics() {
+            deque.extend(generics);
+        }
+    }
```

Comment:
Nit: I don't really see `VecDeque`'s API being used, here, so let's keep it simple and use a `Vec` 🙂 

```suggestion
    let mut stack: Vec<&Type> = trait_.iter().collect();
    while let Some(ty) = stack.pop() {
        if let Some(did) = ty.def_id() {
            if cx.tcx.get_attrs(did).lists(sym::doc).has_word(sym::hidden) {
                return;
            }
        }
        if let Some(generics) = ty.generics() {
            stack.extend(generics);
        }
    }
```

Reply:
Done